### PR TITLE
Better recovery for invalid URLs and content types

### PR DESCRIPTION
**Pull request by spericas**
See Issue #1218. Return HTTP 400 when appropriate; there are some cases now where we either return an empty response or a 500 error. Exceptions are still logged for further analysis.

### CODE REVIEW

File: webserver/webserver/src/main/java/io/helidon/webserver/RequestRouting.java [hunk 1]
```diff
@@ -380,6 +380,10 @@ private void defaultHandler(Throwable t) {
 
                     response.status(Http.Status.INTERNAL_SERVER_ERROR_500);
                 }
+                String message = t.getMessage();
+                if (message != null) {
+                    response.send(message);
```
Comment (romain-grecourt) [inline webserver/webserver/src/main/java/io/helidon/webserver/RequestRouting.java]: How does that play with the response.send() below ?

Comment (spericas) [reply webserver/webserver/src/main/java/io/helidon/webserver/RequestRouting.java]: Forgot to add WIP to the title. But that needs to be fixed.

Comment (romain-grecourt) [reply webserver/webserver/src/main/java/io/helidon/webserver/RequestRouting.java]: Oh ok, no problem.

File: webserver/webserver/src/main/java/io/helidon/webserver/ForwardingHandler.java [hunk 2]
```diff
@@ -138,7 +138,13 @@ protected void channelRead0(ChannelHandlerContext ctx, Object msg) {
                 send100Continue(ctx);
             }
 
-            routing.route(bareRequest, bareResponse);
+            // If a problem during routing, return 400 response
+            try {
+                routing.route(bareRequest, bareResponse);
+            } catch (IllegalArgumentException e) {
```
Comment (tjquinno) [inline webserver/webserver/src/main/java/io/helidon/webserver/ForwardingHandler.java]: Just wondering... Are there other exceptions besides `IllegalArgumentException` that might be thrown that should also trigger the bad request response? I see that the checking at line 111 is the same.

Comment (spericas) [reply webserver/webserver/src/main/java/io/helidon/webserver/ForwardingHandler.java]: I haven't encounter other exceptions, but perhaps. This is intended to be an improvement over what we have now, but perhaps not the last one of these.

**Pull request merged**